### ISSUE TITLE
Use pool as a queue

### ISSUE THREAD

**Issue by ffb255** [panjf2000/ants#76]
**Is your feature request related to a problem? Please describe.**
I have a limited pool (sized pool), but when I reach the capacity of the pool, `Submit()` returns an error (`ErrPoolOverload`) which means the pool is full and no workers available.
Even if i set `Nonblocking` to true in [`Option`].(https://gowalker.org/github.com/panjf2000/ants#Options)
```Go
func main() {
	p, _ := ants.NewPool(2, ants.WithNonblocking(true))

	for i := 0; i < 5; i++ {
		err := p.Submit(func() {
			fmt.Println("Testing wait for 1 second...")
			time.Sleep(1 * time.Second)
		})
		if err != nil {
			fmt.Println("Error: " + err.Error())
		}
	}

	time.Sleep(500 * time.Second) // Only for test and block main(), something like waitgroup!
}
```
This code returns:
```
Error: too many goroutines blocked on submit or Nonblocking is set
Error: too many goroutines blocked on submit or Nonblocking is set
Error: too many goroutines blocked on submit or Nonblocking is set
Something is here, waiting..
Something is here, waiting..
```

**Describe the solution you'd like**
I think it's better to have something like a (second) queue in ants. tasks submit without any error and complete one-by-one.
In other words, `Submit()` should be non-blocking but it shouldn't return `ErrPoolOverload` error even if capacity is full, and should continue receiving tasks.
I'm not sure if this feature exists in ants or not.

**Describe alternatives you've considered**

**Additional context**
`go` keyword can do that:
```Go
p, _ := ants.NewPool(2, ants.WithNonblocking(false))

for i := 0; i < 5; i++ {
    go p.Submit(func() {
        fmt.Println("Testing wait for 1 second...")
        time.Sleep(1 * time.Second)
    })
}
```
But i think there must be a better way.

**Comment by panjf2000** [comment]
It makes no sense to have such a queue inside `ants` cuz you were submitting tasks to a queue in a way so there is no need to add another queue. If you are worried about that capacity might be too small to receive your tasks, just increase the pool size and it will work.

Besides, let's say we do add one more queue you are requesting, but that queue also need to set a size inside, How much capacity do you want to set for it and what does `ants` do when you reach the capacity of the new queue again? returns nil or be blocked? Aren't we back to the original problem again?

**Comment by panjf2000** [comment]
#56

**Comment by ffb255** [comment]
(first post updated, template added)
I saw your diagram (README) and realized how `ants` works.

> If you are worried about that the capacity of pool might be too small to receive your tasks, just increase the pool size and it will work.
Consider i've 20,000 heavy mathematics calculation and for a better performance, I have to do them two by two. So i should set the pool size to 2.
Furthermore, As I said i've 20,000 tasks and i need to add all of them to queue at the same time.
In the current version i can't bulk add them to queue and i should wait for task completion to add new task.

**Comment by vtolstov** [comment]
interesting feature. so we have something like number of concurrent workers and pool size with number of items in wait state

**Comment by HaoKunT** [comment]
According to the author's idea, `ants` only provide a pool, the queue should be written by the user, or use another package.

**Comment by zichenglang** [comment]
@ffb255 
antspool is a goroutine worker pool, you should manage jobs by yourself, such as using queue to store them and pop them into worker pool.

 I think this maybe what you need.
https://github.com/lacdon/gopool

**Comment by ffb255** [comment]
@zichenglang After opening this issue, I wrote my own in-memory queue which works inside `ants`.

**Comment by zicheqingluo** [comment]
同意[ffb255]观点，如果设置协程池为2，而我们有10个任务，必须等待当前两个执行完才能submit后面的，那协程池的意义是不是不存在呢，协程不就是想不阻塞当前，可现在他阻塞了。

**Comment by zicheqingluo** [comment]
所以，我为什么要用ants呢，啥都没有
